CloudAppSecurityDocs/api-data-enrichment-update.md

```diff
@@ -18,13 +18,17 @@ POST /api/v1/subnet/<ip_range_id>/update_rule/
 
 ## Request BODY parameters
 
-| Parameter | Description |
-| --- | --- |
-| name | The unique name of the range |
-| category | The id of the range category. Providing a category helps you easily recognize activities from interesting IP addresses. Possible values include:<br /><br />**1**: Corporate<br />**2**: Administrative<br />**3**: Risky<br />**4**: VPN<br />**5**: Cloud provider<br />**6**: Other |
-| subnets | An array of masks as strings (IPv4 / IPv6) |
-| organization (Optional) | The registered ISP |
-| tags (Optional) | An array of new or existing objects including the tag name, id, description, name template, and tenant id |
+| Parameter | Type | Description |
+| --- | --- | --- |
+| name | string | The unique name of the range |
+| _id (Optional)| string | The IP range id. Required if you are not changing the name. |
```

Comment:
remove the _id parameter as it is used from the URL only

Reply:
Hey @dagoltz,  this does not appear to be the case for update calls where you are not trying to update the name. Please see the below sample calls/returns.

All calls are POST requests to `https://tenantname.us3.portal.cloudappsecurity.com/api/v1/subnet/<ip_range_id>/update_rule/` with valid credentials.

Case 1, Update subnets without specifying the `name` parameter in the body:
```json
{
  "category":5,
  "subnets":[
    "192.168.1.0/24",
    "192.168.2.0/16"
  ]
}
```
Result:
```json
{
    "errors": [
        {
            "error": "Missing name for the subnet range rule.",
            "errorMessageCode": "CONSOLE_IP_RULE_ERROR_MISSING_NAME_FOR_THE_SUBNET_RULE"
        }
    ]
}
```

Case 2, Update subnets with specifying the `name` parameter in the body:
```json
{
  "name":"testRange",
  "category":5,
  "subnets":[
    "192.168.1.0/24",
    "192.168.2.0/16"
  ]
}
```
Result:
```json
{
    "errors": [
        {
            "error": "A range with the same name already exists.",
            "errorMessageCode": "CONSOLE_IP_RULE_ERROR_DUPLICATE_NAME_FOR_THE_SUBNET_RULE"
        }
    ]
}
```

Case 3, Update subnets with specifying the `name` parameter and `_id` in the body:
```json
{
  "name":"testRange",
  "_id":"<ip_range_id>",
  "category":5,
  "subnets":[
    "192.168.1.0/24",
    "192.168.2.0/16"
  ]
}
```
Result:
```json
{
    "name": "testRange",
    "subnets": [
        {
            "mask": 120,
            "address": "0000:0000:0000:0000:0000:ffff:c0a8:0100",
            "originalString": "192.168.1.0/24"
        },
        {
            "mask": 112,
            "address": "0000:0000:0000:0000:0000:ffff:c0a8:0000",
            "originalString": "192.168.2.0/16"
        }
    ],
    "location": null,
    "organization": null,
    "tags": [],
    "category": 5,
    "lastModified": 1633375827168.9678
}
```

Case 4, Update subnets and `name` without specifying the `_id` parameter in the body:
```json
{
  "name":"testRange2",
  "category":5,
  "subnets":[
    "192.168.1.0/24",
    "192.168.2.0/16"
  ]
}
```
Result:
```json
{
    "name": "testRange2",
    "subnets": [
        {
            "mask": 120,
            "address": "0000:0000:0000:0000:0000:ffff:c0a8:0100",
            "originalString": "192.168.1.0/24"
        },
        {
            "mask": 112,
            "address": "0000:0000:0000:0000:0000:ffff:c0a8:0000",
            "originalString": "192.168.2.0/16"
        }
    ],
    "location": null,
    "organization": null,
    "tags": [],
    "category": 5,
    "lastModified": 1633376093881.685
}
```

Reply:
Hi @mcvic1rj ,
the API is built to accept all fields at all time, it is not built for partial updates.
so as the name property is required, any update without it (even if you add the not-used "_id" property) will be responded with an error.

partial updates will usually be done using PATH or PUT HTTP verbs, which we do not support for this endpoint.